src/main/java/no/fint/apistatus/service/HealthRepository.java

```diff
@@ -0,0 +1,47 @@
+package no.fint.apistatus.service;
+
+import lombok.Data;
+import no.fint.apistatus.model.HealthCheckResponse;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+@Data
+@Component
+public class HealthRepository {
+
+    private final ConcurrentMap<String, List<HealthCheckResponse>> healthChecks = new ConcurrentHashMap<>();
```

Comment:
Since you're adding `Collection.synchronizedList()` for every map entry, the benefit of using a concurrent map is minimal.  Using a synchronized `MultiMap` is probably just as fast, and will most likely result in simpler code.

Reply:
fixed